config/config.go

```diff
@@ -68,6 +68,41 @@ func New(uri string, requests, concurrency int, requestTimeout, globalTimeout ti
 	}
 }
 
+// Validate returns the config and a not nil ErrInvalid if any of the fields provided by the user is not valid
+func (cfg Config) Validate() error {
+	inputErrors := []error{}
+
+	_, err := url.ParseRequestURI(cfg.Request.URL.String())
+	if err != nil {
+		inputErrors = append(inputErrors, fmt.Errorf("-url: %s is not a valid url", cfg.Request.URL.String()))
+	}
+
+	if cfg.RunnerOptions.Requests < 0 {
+		if cfg.RunnerOptions.Requests != -1 {
+			inputErrors = append(inputErrors, fmt.Errorf("-requests: must be >= 0, we got %d", cfg.RunnerOptions.Requests))
+		}
+	}
+
+	if cfg.RunnerOptions.Concurrency < 0 {
+		if cfg.RunnerOptions.Concurrency != -1 {
```

Comment:
Actually I realize we probably shouln't accept any value below `1` for `concurrency`, because `1` already stands for "no concurrency" 🤔

Reply:
Indeed!